docs/detailed-push-process.md

```diff
@@ -0,0 +1,85 @@
+# Epinio push in detail
+
+Epinio strives to make use of well supported, well known and loved projects instead of re-inventing the wheel ([link](README.md#guidelines-soft-principles)).
+But while doing so, it makes sure those components are deployed correctly and work together seamlessly. Let's go through the `epinio push` process in detail
+so you can understand what each components does.
+
+You can see an image that visualises the process lower in this page. Refer to it while reading the text to help you understand the process more.
+
+## 1. Uploading the code
+
+One of the components Epinio installs on your cluster is [Gitea](https://gitea.io/en-us/). Gitea is an Open Source code hosting solution. Among other things it allows
+us to create repositories and organizations using API calls. It also used to store your application's code using which Epinio pushes using [`git`](https://git-scm.com/).
```

Comment:
`It also` / `It is also`

`using which Epinio pushes using`
`which Epinio pushes using`